File: RepoTransBench/target_projects/Python/Go/RicherMans_PLDA/tests/original/pldatest_test.go
package original

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"richermans_plda/liblda"
)

func TestPLDAFit(t *testing.T) {
	X := [][]float64{
		{1, 2},
		{3, 4},
		{5, 6},
	}
	y := []int{0, 1, 0}
	plda := liblda.NewPLDA()
	out := plda.Fit(X, y)
	assert.True(t, plda.Trained)
	assert.Equal(t, plda, out)
}

func TestPLDAPredict(t *testing.T) {
	X := [][]float64{
		{1, 2},
		{3, 4},
		{5, 6},
	}
	y := []int{0, 1, 0}
	plda := liblda.NewPLDA()
	plda.Fit(X, y)
	pred := plda.Predict(X)
	for _, p := range pred {
		assert.Equal(t, 0, p)
	}
}

func TestPLDAPredictWithoutFit(t *testing.T) {
	X := [][]float64{
		{1, 2},
		{3, 4},
		{5, 6},
	}
	p := liblda.NewPLDA()
	_, err := p.PredictWithErr(X)
	assert.Error(t, err)
}

File: RepoTransBench/target_projects/Python/C++/RicherMans_PLDA/tests/original/pldatest_test.cpp
#include <gtest/gtest.h>
#include "src/liblda/plda.h"

TEST(TestPLDA, Fit) {
    std::vector<std::vector<double>> X{{1,2},{3,4},{5,6}};
    std::vector<int> y{0,1,0};
    PLDA plda;
    PLDA& out = plda.fit(X, y);
    ASSERT_TRUE(plda.trained);
    ASSERT_EQ(&out, &plda);
}

TEST(TestPLDA, Predict) {
    std::vector<std::vector<double>> X{{1,2},{3,4},{5,6}};
    std::vector<int> y{0,1,0};
    PLDA plda;
    plda.fit(X, y);
    std::vector<int> pred = plda.predict(X);
    // pred vector should all be zero
    for (auto v : pred) {
        ASSERT_EQ(v, 0);
    }
}

TEST(TestPLDA, PredictWithoutFit) {
    std::vector<std::vector<double>> X{{1,2},{3,4},{5,6}};
    PLDA p;
    try {
        p.predict(X);
        FAIL() << "Expected std::runtime_error";
    } catch (const std::runtime_error& e) {
        SUCCEED();
    } catch (...) {
        FAIL() << "Expected std::runtime_error";
    }
}